Graphics/Primitives/IPixelatedPrimitiveRenderer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+
+namespace CalamityMod.Graphics.Primitives
+{
+    /// <summary>
+    /// Use to sucessfully render primitives with pixelation with an NPC or Projectile.
+    /// </summary>
+    public interface IPixelatedPrimitiveRenderer
+    {
+        /// <summary>
+        /// The layer to render the primitive(s) to.
+        /// </summary>
+        public PixelationPrimitiveLayer LayerToRenderTo => PixelationPrimitiveLayer.BeforeProjectiles;
+
+        /// <summary>
+        /// Render primitives that use pixelation here.
+        /// </summary>
+        public void RenderPixelatedPrimitives(SpriteBatch spriteBatch);
```

Comment:
Do we define access modifiers for other interfaces in Cal? Can we stop doing that if so?

Reply:
Whats the reason they don't have them? I know they can only be public, but it seems weird to me to not include them on interfaces only.